### PR TITLE
Switch canonical json

### PR DESCRIPTION
**Pull request by carolynvs**
Per CNAB Spec cnabio/cnab-spec#414, we want to support numbers in our canonical json representation.

The library we are currently using, github.com/docker/go/json, does not support this. So I am migrating us to the library mentioned in the spec as being compliant. This is the same library used by cnab-go in https://github.com/cnabio/cnab-go/pull/247.

I was not able to completely remove the import of the old library because TUF uses its RawMessage struct, which is a very simple wrapper around a byte array.

If we are interested we can try to get TUF to use an interface instead of a hard-coded struct type so that we can drop the dependency on the other canonical json library.

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
Interesting! Let's discuss at the CNAB Security meeting this week please...

**Comment by carolynvs** [comment]
I've added it to Wednesday's agenda.

**Comment by radu-matei** [comment]
FYI, the expiration error is now fixed by #91.

Happy to chat about this at the next meeting, but as long as _all_ CNAB tooling has the same serialization rules, it should not affect how we sign and verify bundles, although keep in mind this is a breaking change, i.e. the same bundle _could_ get a different content digest and invalidate the signature.

**Comment by carolynvs** [comment]
@trishankatdatadog @radu-matei I've rebased to pick up the fix from main and made the linter happy.

File: pkg/canonicaljson/canonicaljson.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	canonicaljson "github.com/cyberphone/json-canonicalization/go/src/webpki.org/jsoncanonicalizer"
+	rawmessage "github.com/docker/go/canonical/json" // We are only using this library for the RawMessage type that TUF uses, the actual marshaling is done by the webpki.org/jsoncanonicalizer library
+)
+
+// define a type alias for raw message in this package
+// so that outside of this package we only import a single canonical json library.
+type RawMessage = rawmessage.RawMessage
+
+// Marshal returns the canonical json encoded value of v.
+func Marshal(v interface{}) ([]byte, error) {
```
Comment (trishankatdatadog) [inline pkg/canonicaljson/canonicaljson.go:15]: Why do we need this function if we have `MarshalToRawMessage` later which checks for errors? Do we not want to check for errors all the time?

Comment (carolynvs) [reply pkg/canonicaljson/canonicaljson.go:15]: I just pushed a commit removing MarshalToRawMessage.

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**